core/src/test/java/com/linecorp/armeria/common/stream/PathStreamMessageTest.java

```diff
@@ -212,6 +212,97 @@ void nullBlockingTaskExecutor_withClientRequestContext() {
+    }
+
+    @Test
+    void skip0_take13() {
+        final Path path = Paths.get("src/test/resources/com/linecorp/armeria/common/stream/test.txt");
+        final ByteStreamMessage publisher = StreamMessage.builder(path).bufferSize(1).build()
+                                                         .range(0, 13);
+        final StringAggregator stringAggregator = new StringAggregator();
+
+        publisher.subscribe(stringAggregator);
+        final String result = stringAggregator.future.join();
+
+        assertThat(result).isEqualTo("A1234567890\nB");
+    }
+
+    @Test
+    void skip12() {
+        final Path path = Paths.get("src/test/resources/com/linecorp/armeria/common/stream/test.txt");
+        final ByteStreamMessage publisher = StreamMessage.builder(path).bufferSize(1).build()
+                                                         .range(12, Long.MAX_VALUE);
+        final StringAggregator stringAggregator = new StringAggregator();
+
+        publisher.subscribe(stringAggregator);
+        final String result = stringAggregator.future.join();
+
+        assertThat(result).isEqualTo("B1234567890\nC1234567890\nD1234567890\nE1234567890\n");
+    }
+
+    @Test
+    void skip12_take13() {
+        final Path path = Paths.get("src/test/resources/com/linecorp/armeria/common/stream/test.txt");
+        final ByteStreamMessage publisher = StreamMessage.builder(path).bufferSize(1).build()
+                                                         .range(12, 13);
+        final StringAggregator stringAggregator = new StringAggregator();
+
+        publisher.subscribe(stringAggregator);
+        final String result = stringAggregator.future.join();
+
+        assertThat(result).isEqualTo("B1234567890\nC");
+    }
+
+    @Test
+    void skipAll() {
+        final Path path = Paths.get("src/test/resources/com/linecorp/armeria/common/stream/test.txt");
+        final ByteStreamMessage publisher = StreamMessage.builder(path).bufferSize(1).build()
+                                                         .range(999, Long.MAX_VALUE);
+        final StringAggregator stringAggregator = new StringAggregator();
+
+        publisher.subscribe(stringAggregator);
+        final String result = stringAggregator.future.join();
+
+        assertThat(result).isEmpty();
+    }
+
+    @Test
+    void skipMost() {
+        final Path path = Paths.get("src/test/resources/com/linecorp/armeria/common/stream/test.txt");
+        final ByteStreamMessage publisher = StreamMessage.builder(path).bufferSize(1).build()
+                                                         .range(48, Long.MAX_VALUE);
+        final StringAggregator stringAggregator = new StringAggregator();
+
+        publisher.subscribe(stringAggregator);
+        final String result = stringAggregator.future.join();
+
+        assertThat(result).isEqualTo("E1234567890\n");
+    }
+
+    @Test
+    void skip1() {
```

Comment:
```
AS-IS: end = offset + length;
TO-BE: end = LongMath.saturatedAdd(offset, length);
```
https://github.com/line/armeria/blob/bc48105148d652e2e48e788c94bec007882c31dc/core/src/main/java/com/linecorp/armeria/common/stream/PathStreamMessage.java#L343

```
14:38:39.865 [armeria-common-worker-nio-2-9] WARN  i.n.u.c.AbstractEventExecutor - A task raised an exception. Task: com.linecorp.armeria.common.stream.PathStreamMessage$PathSubscription$$Lambda$544/0x0000000800354840@4bb4df46
java.lang.AssertionError: null
	at com.linecorp.armeria.common.stream.PathStreamMessage$PathSubscription.lambda$completed$0(PathStreamMessage.java:343)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
    ...
```
(FYI) On `AS-IS`,  `skip1` test failed with above error due to overflow.
`assert position(0) <= end(-XXX, overflow)` assertion failed~!

Reply:
Thanks for the fix, @injae-kim !